### PR TITLE
add dropdown menu display mode

### PR DESCRIPTION
**Pull request by shaneahern**
- add optional prop displayAsDropdown default false to allow display of input as a dropdown menu instead of a VAS slider

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -91,8 +96,23 @@ export default class VASSlider extends React.PureComponent {
     } = this.props;
     const { rating } = this.state;
     const { head, tail } = this.getRateValues();
+    const dropdownSource = Array.from(
+      Array((max - min) + 1), (e, i) =>
```
Comment (slycoder) [inline src/index.js:100]: The problem is that min/max is not all there is to it; there's also rate values. 

I would recommend relying on getRateValues() to figure out the correct ordering, and then using translateRating/Position above to ensure that if the orientation of the values needs to be flipped (i.e. selecting a 3 should return a 7) that'll happen.

Comment (shaneahern) [reply src/index.js]: added a new function dropdownRatingSource that uses translatePosition/getRateValues to compute the display values based on a 0.0 to 1.0 range for ratings with step of 0.1 to map to the display values of 0 to 10

File: src/index.js [hunk 2]
```diff
@@ -81,6 +84,20 @@ export default class VASSlider extends React.PureComponent {
...
+    const { min, max } = this.props;
+    const start = 0;
+    const end = 1;
+    const step = end / (max - min);
+    const len = Math.floor((end - start) / step) + 1;
+    const ratingVals = Array(len).fill().map(
+        (_, idx) => Number((start + (idx * step)).toFixed(1)),
+    );
+    return Array.from(ratingVals, rating =>
+        ({ value: rating, label: this.translatePosition(rating) }),
```
Comment (slycoder) [inline src/index.js:97]: Might be worth adding the text to the endpoints (see head/tail.text).

File: src/index.js [hunk 3]
```diff
@@ -81,6 +84,20 @@ export default class VASSlider extends React.PureComponent {
     return (rating - head.value) / (tail.value - head.value);
   }
 
+  dropdownRatingSource = () => {
+    const { min, max } = this.props;
+    const start = 0;
+    const end = 1;
+    const step = end / (max - min);
+    const len = Math.floor((end - start) / step) + 1;
+    const ratingVals = Array(len).fill().map(
```
[the remark this answers is not in the document]
Comment (shaneahern) [reply src/index.js:93]: was wanting to use lodash function but wasn't sure if we want to introduce another library into the project -- might as well add it and might be some other places we can use it

File: src/index.js (hunk 2 above)
Comment (shaneahern) [reply src/index.js]: done

**Comment by shaneahern** [comment]
What's the setup for reversing the order of the display values, is it reversing the min and max props? e.g. min = 10 max = 0?

**Comment by slycoder** [comment]
It uses rate values.  Here's the spec: https://www.notion.so/physera/VAS-Slider-7c29da7249f84353ae052f865b3b22e5

**Comment by shaneahern** [comment]
Verifed that the dropdown outputs the correct rating value to onRatingChange when in reverse order (10 to 0 vs 0 to 10), mergeing

**Pull request merged**